app/src/main/java/com/github/polypoly/app/game/PendingGame.kt

```diff
@@ -0,0 +1,64 @@
+package com.github.polypoly.app.game
+
+import kotlin.time.Duration
+
+class PendingGame(
+    val admin: User,
+    val gameMode: GameMode,
+    val minimumNumberOfPlayers: Int,
+    val maximumNumberOfPlayers: Int,
+    val roundDuration: Duration,
+    val gameMap: List<Zone>,
+    val initialPlayerBalance: Int,
+    val name: String,
+    val code: String
+) {
+
+    private val currentUsersRegistered: ArrayList<User> = ArrayList()
+    val usersRegistered: List<User> get() = usersRegistered.toList()
+
+    init {
+        if (minimumNumberOfPlayers < 1)
+            throw java.lang.IllegalArgumentException("At least 2 players are needed for a game (provided $minimumNumberOfPlayers)")
+        if (maximumNumberOfPlayers < minimumNumberOfPlayers)
+            throw java.lang.IllegalArgumentException("Maximum number of players $maximumNumberOfPlayers must be greater than the minimum number $minimumNumberOfPlayers")
+        if (roundDuration.isNegative() || roundDuration.isInfinite())
+            throw java.lang.IllegalArgumentException("Invalid game duration$roundDuration")
+        if (name.isEmpty())
+            throw java.lang.IllegalArgumentException("Game name cannot be empty")
+        if (name.isBlank())
+            throw java.lang.IllegalArgumentException("Game name cannot be blank")
+        if (code.isEmpty())
+            throw java.lang.IllegalArgumentException("Game code cannot be empty")
+        if (code.isBlank())
+            throw java.lang.IllegalArgumentException("Game code cannot be blank")
+    }
+
+    fun addUser(user: User) {
```

Comment:
I hesitated on that one but I thought that as the number of players is publicly available, it should be the caller's responsability (and more specifically probably in our case the joining activity) to ensure that no user attemps to join a game already full. This is the reason why I find it more logic to response to such incorrect behavior with an exception rather than a boolean.
It might be more efficient and less verbose to use a boolean instead when this will actually be used, however I still believe it makes more sense as it is for now :)